s3-native-client/src/generated/java/com/amazonaws/s3/model/GetObjectRequest.java

```diff
@@ -564,6 +591,16 @@ public final Builder expectedBucketOwner(String expectedBucketOwner) {
             return this;
         }
 
+        public final Builder customHeaders(HttpHeader[] customHeaders) {
+            this.customHeaders = customHeaders;
+            return this;
+        }
+
+        public final Builder customQueryParameters(String customQueryParameters) {
```

Comment:
Would it make sense to allow users to provide a single query parameter instead? Otherwise users would need to build the query parameters by themselves. 

This is Java SDK v2's API
https://github.com/aws/aws-sdk-java-v2/blob/master/core/sdk-core/src/main/java/software/amazon/awssdk/core/RequestOverrideConfiguration.java#L254-L256

Reply:
Hmmm I do like that idea.  But, I think we'd have to encode the query parameters, and I'm not sure we have a Base64 encoder we can use inside of the Java CRT right now.  I'll double check!

Reply:
I guess we can require the customer to have properly prepared the parameters and still build the actual string for them.  I'll look at doing that.

Reply:
Changed it to a Map<String,String>.